src/main/java/teammates/ui/webapi/output/InstructorPrivilegeData.java

```diff
@@ -0,0 +1,59 @@
+package teammates.ui.webapi.output;
+
+import java.util.Map;
+
+import teammates.common.datatransfer.InstructorPrivileges;
+
+/**
+ * The output format of instructor privileges.
+ */
+public class InstructorPrivilegeData extends ApiOutput {
+    private InstructorPrivileges privileges;
```

Comment:
I don't think you should expose `InstructorPrivileges`. It is kind of internal data structure to store privilege of an instructor (i.e. map).

![image](https://user-images.githubusercontent.com/13977235/53332520-84c08880-392e-11e9-8a93-60bb86e9955a.png)

This is what you should put in the response DTO. The structure of the DTO will the same while it changed for different granularity (course, section, feedbackSession).

Reply:
Sorry, you are saying we should wrap the `InstructorPrivileges` to make the response DTO(still a map)? 

I just used `InstructorPrivileges` as  a container, which contains 3 maps(courseLevel, sectionLevel, sessionLevel), and put privileges as required into the corresponding map. 
From my understanding, those privileges are quite self-explanatory from what I saw from the old endpoints. For example,
```
        response.setCanModifyCourse(
                instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE));
```
I did not say any translation needed here.

So is it okay to just create my own version of container of 3 maps and use it to substitute `InstructorPrivileges`?

Reply:
Yup. A map is fine but don't use `InstructorPrivileges` for convenience.